src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,1063 @@
+! stdlib_stringlist.f90 --
+!     Module for storing and manipulating list of strings
+!     The strings may have arbitrary lengths, not necessarily the same
+!
+!     Note: very preliminary
+!
+!     TODO:
+!     insert( list_end, ... ) in an empty list?
+!     concatenate two string lists
+!
+!     Not implemented yet:
+!     insert a list or an array of character strings
+!     replace a string, list or an array of character strings
+!     concatenate a list with another list or an array
+!
+!     Limited to implemented routines
+!
+module stdlib_stringlist
+    use stdlib_string_type, only: string_type, move, assignment(=) !, char, operator(==)
```

Comment:
move: `move_alloc` for `string_type` find it [here](https://github.com/fortran-lang/stdlib/pull/467)